File: Assignment 12/Tony Gradis Java/8.8 Parking Ticket Simulator/Code/PoliceOfficer.java
package Code;

public class PoliceOfficer {
    private String name;
    private String badgeNumber;

    public PoliceOfficer(String name, String badgeNumber) {
        this.name = name;
        this.badgeNumber = badgeNumber;
    }

    public ParkingTicket inspectCar(ParkedCar car, ParkingMeter meter) {
        int overParkedMinutes = car.getParkedMinutes() - meter.getPurchasedMinutes();
        if (overParkedMinutes > 0) {
            return new ParkingTicket(car, name, badgeNumber, overParkedMinutes);
        }
        return null;
    }
}

File: Assignment 12/Tony Gradis C++/14.14 Parking Ticket Simulator/Code/PoliceOfficer.cpp
#include "PoliceOfficer.h"

PoliceOfficer::PoliceOfficer(const string& officerName, const string& badge) : name(officerName), badgeNumber(badge) {}

ParkingTicket* PoliceOfficer::inspectCar(const ParkedCar& car, const ParkingMeter& meter) const {
    int overParkedMinutes = car.getParkedMinutes() - meter.getPurchasedMinutes();
    if (overParkedMinutes > 0) {
        return new ParkingTicket(car, name, badgeNumber, overParkedMinutes);
    }
    return nullptr;
}